src/main/scala/com/databricks/spark/redshift/RedshiftWriter.scala

```diff
@@ -311,29 +311,25 @@ private[redshift] class RedshiftWriter(
       // for a description of the manifest file format. The URLs in this manifest must be absolute
       // and complete.
 
-      // The saved filenames are going to be of the form part-r-XXXXX-UUID.avro. There isn't a way
-      // to figure out what this UUID is, besides querying S3. In principle, this file format could
-      // change in a way that breaks this library, but this seems unlikely and we can patch around
-      // it if that happens.
+      // The saved filenames depend on the spark-avro version. In spark-avro 1.0.0, the write
+      // path uses SparkContext.saveAsHadoopFile(), which produces filenames of the form
+      // part-XXXXX.avro. In spark-avro 2.0.0+, the partition filenames are of the form
+      // part-r-XXXXX-UUID.avro.
       val fs = FileSystem.get(URI.create(tempDir), sqlContext.sparkContext.hadoopConfiguration)
-      val writeJobUUID = fs.listStatus(new Path(tempDir))
```

Comment:
In terms of efficiency: note that here we call `fs.listStatus()` on the root of the write-specific `tempDir` and we make the same `listStatus` call in the updated code below; the only change is that we now filter this list rather than trying to extract the UUID and use that to construct the right filenames.

Both versions of the code potentially run into problems when listing directories with millions of files; for this reason, I'm wondering whether I should use the Amazon S3 client to perform this listing rather than using the FileSystem API.